From 852df5857052f8748429cf4e04aa18cc5cce36e6 Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Fri, 31 Mar 2023 00:08:44 +0200
Subject: [PATCH] ashmem: Prevent integer overflow protection
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This gets triggered by Dolphin Emu when using memfd.
Works fine with just ignoring ¯\_(ツ)_/¯

Change-Id: I8082037ae7cfce2e94917ce1786137f0f6e6cf13
---
 libcutils/ashmem-dev.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libcutils/ashmem-dev.cpp b/libcutils/ashmem-dev.cpp
index 8c232f0cd..297cf8629 100644
--- a/libcutils/ashmem-dev.cpp
+++ b/libcutils/ashmem-dev.cpp
@@ -447,6 +447,7 @@ int ashmem_unpin_region(int fd, size_t offset, size_t len)
     return __ashmem_check_failure(fd, TEMP_FAILURE_RETRY(ioctl(fd, ASHMEM_UNPIN, &pin)));
 }
 
+__attribute__((no_sanitize("integer")))
 int ashmem_get_size_region(int fd)
 {
     if (has_memfd_support() && !memfd_is_ashmem(fd)) {
-- 
2.39.2

